### DISCUSSION TITLE
Detecting scroll up

### DISCUSSION THREAD

**Discussion by lucperkins** [alpinejs/alpine#1513] [1. Help]
Do you any of you have a preferred way of detecting an upward scroll in Alpine.js? I have a few ways that I've tried out that feel really kludgey and I was hoping that maybe someone in the community has come up with an elegant way of doing that.

**Comment by KevinBatdorf** [comment]
Maybe you could store the position in `x-data` then have a scroll event that checks whether it's going up? 

Share what you got so far. It's probably fine.

**Reply by lucperkins** [reply]
I actually decided against a scroll up detector for my use case here, *but* I do think that this might be a nice built-in!

```html
<body x-data="{ showNav: false }" @scroll.up="showNav = true">
  <nav x-show="showNav">
    <!-- nav content -->
  </nav>

  <!-- page content -->
</body>
```

I'll consider filing an issue.

**Reply by danddanddand** [reply]
I've been using @scroll.window="yourHandleScrollLogic()" for our mobile/normal menu to make it sticky. It's probably one of the earliest alpine components we built so have not looked at it in long time and no idea if it's best practice.
